### DISCUSSION TITLE
[find] Query with filtering by input a string value for the field that is an integer type

### DISCUSSION THREAD

**Discussion by codyng** [feathersjs/feathers#3212] [General]
Please try:

1. A model say "submissions" with one of the field called "year" (positive integer type). (Others fields can be name, email, etc)
2. Enter some rows to the DB.
3. Make a endpoint that accepts the [find] and query parameters like name, email, year, etc.
4. Your endpoint should accept the string type for query in case the client side passes string type to the year, e.g. a100. You may redefine the schema for query to accept string value for the year field.
5. All records with year valued 0 are returned for the [find] endpoint if the year value is not an integer type. Suppose empty records are returned.

Not sure if it is Knex or Feathers problem. The a100 is treated to be 0 (integer) in the query object:
query: {
    '$skip': 0,
    '$limit': 20,
    year: 'a100'
}

**Comment by daffl** [comment]
The v5 [query schema validators](https://feathersjs.com/api/schema/validators.html) should catch this and only coerce valid numbers.
